### PR TITLE
NEW: Fixes #69 With configurable "api fields" for related objects

### LINKED ISSUE #69: Allow additional config driven fields in JSONDataFormatter
In `JSONDataFormatter` line 119 and 151 a baked-in set of three fields are returned for each object in a has_xxx relation which doesn't allow for much in the way of customisation, viz:

```
                $serobj->$relName = ArrayData::array_to_object(array(
                    "className" => $relClass,
                    "href" => "$href.json",
                    "id" => $obj->$fieldName
                ));
```

We should allow developers to specify additional fields to return here, declared in YML config viz:

```
                $serobj->$relName = ArrayData::array_to_object(array_replace([
                    "className" => $relClass,
                    "href" => "$href.json",
                    "id" => $obj->$fieldName
                ], $obj->config()->get('api_fields'));
```

Note the use of `array_replace()` so that userland config can also _override_ the defaults, not simply append to them.

### PR DESCRIPTION
**Pull request by phptek**

### CODE REVIEW

File: src/DataFormatter/JSONDataFormatter.php [hunk 1]
```diff
@@ -118,10 +118,14 @@ public function convertDataObjectToJSONObject(DataObjectInterface $obj, $fields
...
                 $href = Director::absoluteURL($rel);
-                $serobj->$relName = ArrayData::array_to_object(array(
+                $baseFields = [
                     "className" => $relClass,
                     "href" => "$href.json",
                     "id" => self::cast($obj->obj($fieldName))
+                ];
+                $serobj->$relName = ArrayData::array_to_object(array_replace(
+                    $baseFields,
+                    method_exists($serobj, 'getApiFields') ? (array) $serobj->getApiFields($baseFields) : []
```
Comment (robbieaverill) [inline src/DataFormatter/JSONDataFormatter.php]: Do you want to use `hasMethod()` via ClassInfo here, which would also include extensions that provide this method?

Comment (phptek) [reply src/DataFormatter/JSONDataFormatter.php]: Nice...

Comment (phptek) [reply src/DataFormatter/JSONDataFormatter.php]: Done.

**Comment by robbieaverill** [comment]
The Travis failures are unrelated to the code changes at this point. I'll fix them tomorrow morning and rebase this PR.

**Comment by phptek** [comment]
All good @robbieaverill - cheers.

**Comment by phptek** [comment]
Anything else needed from me here?

**Comment by NightJar** [comment]
I've restarted the build

**Pull request closed without merging**